numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -353,6 +353,21 @@ gentype_repr(PyObject *self)
     return ret;
 }
 
+static PyObject *
+genint_type_str(PyObject *self)
+{
+    PyObject  *item, *item_str;
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
+    }
+
+    item_str = PyObject_Str(item);
```

Comment:
Hmm maybe that's bad though. I want to preserve the old int-printing behavior as much as possible.

I don't have a 32-bit system to test on, but this means `repr(np.int64(2**63-1))` will now have an extra `L` on that arch, right ?